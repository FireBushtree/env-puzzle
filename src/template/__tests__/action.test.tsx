import * as React from 'react';
import Action from '../action';
import {render, screen} from '@testing-library/react';

const defaultActionProps = {
  selectRows: [],
  dataSource: [],
  columns: [],
  filter: {},
  setColumns: () => {},
  onSelectedRows: () => {},
};

const TABLE_BTN_CLASS = '.env-template-action-btn__table';
const EXPORT_BTN_CLASS = '.env-template-action-btn__export';

describe('<Action />', () => {
  test('default show render all buttons', () => {
    const component = render(<Action {...defaultActionProps} />);
    const {container} = component;
    expect(container).toMatchSnapshot();

    expect(screen.getByText('新建')).toBeTruthy();
    expect(screen.getByText('删除')).toBeTruthy();
    expect(screen.getByText('更多')).toBeTruthy();

    const tableBtn = container.querySelector(TABLE_BTN_CLASS);
    const exportBtn = container.querySelector(EXPORT_BTN_CLASS);

    expect(tableBtn).toBeTruthy();
    expect(exportBtn).toBeTruthy();
  });

  test('hide button', () => {
    const component = render(
      <Action
        {...defaultActionProps}
        createBtn={false}
        exportBtn={false}
        importBtn={false}
        deleteBtn={false}
      />,
    );

    const {container} = component;

    expect(screen.queryByText('新建')).toBeFalsy();
    expect(screen.queryByText('删除')).toBeFalsy();
    expect(screen.queryByText('更多')).toBeFalsy();

    const tableBtn = container.querySelector(TABLE_BTN_CLASS);
    const exportBtn = container.querySelector(EXPORT_BTN_CLASS);

    expect(tableBtn).toBeTruthy();
    expect(exportBtn).toBeFalsy();
  });

  test('moreButtons props will show more button', () => {
    render(
      <Action
        {...defaultActionProps}
        createBtn={false}
        exportBtn={false}
        importBtn={false}
        deleteBtn={false}
        moreButtons={[{name: '绑定信息', onClick: () => {}}]}
      />,
    );

    expect(screen.queryByText('更多')).toBeTruthy();
  });
});
