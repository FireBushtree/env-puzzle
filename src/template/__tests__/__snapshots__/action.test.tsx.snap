// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> default show render all buttons 1`] = `
<div>
  <div
    class="env-template-action"
  >
    <div
      class="env-template-action-left"
    >
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span
          aria-label="plus"
          class="anticon anticon-plus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
        <span>
          新建
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-dangerous"
        type="button"
      >
        <span
          aria-label="delete"
          class="anticon anticon-delete"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="delete"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
            />
          </svg>
        </span>
        <span>
          删除
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary env-template-action-btn__more"
        type="button"
      >
        <span
          aria-label="menu"
          class="anticon anticon-menu"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="menu"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
        <span>
          更多
        </span>
      </button>
    </div>
    <div
      class="env-template-action-right"
    >
      <button
        class="ant-btn env-template-action-btn__table"
        type="button"
      >
        <span
          class="iconfont icon-ai238"
        />
        <span
          aria-label="caret-down"
          class="anticon anticon-caret-down"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="caret-down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
            />
          </svg>
        </span>
      </button>
      <button
        class="ant-btn env-template-action-btn__export"
        type="button"
      >
        <span
          class="iconfont icon-daochu"
        />
        <span
          aria-label="caret-down"
          class="anticon anticon-caret-down"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="caret-down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
